sdk/storage/src/cloud_location.rs

```
@@ -44,20 +92,115 @@ impl CloudLocation {
+                    account,
+                    service_type.subdomain()
+                )
+            }
             CloudLocation::Custom { uri, .. } => uri.clone(),
             CloudLocation::Emulator { address, port } => {
                 format!("http://{address}:{port}/{EMULATOR_ACCOUNT}")
             }
+            CloudLocation::AutoDetect {
+                account,
+                credentials,
+            } => {
+                if let Some(name) = Self::find_cloud_name() {
+                    // These names are from
+                    // `az cloud list --output table`
+                    return match name.as_str() {
+                        // Seems like "AzurePublicCloud" is used in some environments
+                        AZURE_CLOUD | AZURE_PUBLIC_CLOUD => CloudLocation::Public {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        AZURE_US_GOV => CloudLocation::USGov {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        AZURE_CHINA_CLOUD => CloudLocation::China {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        AZURE_GERMAN_CLOUD => CloudLocation::GermanCloud {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        _ => {
+                            todo!()
+                        }
+                    };
+                } else {
+                    // Default to PROD
```

Should we panic here instead to fail fast?